adding ability to receive version report information from clients

josev814 opened the pull request:
adding a requirements txt file

josev814 committed:
adding ability to receive version report information from clients

adding requirements txt file

josev814 committed:
removing check for denyhosts_version

should only be looking for the ip_address to update the client version for

josev814 commented:
I tested this out today with the latest version of denyhosts, and it's working, so this should be good to merge in unless there's something else you want me to do.

I haven't included a report output yet.  I'm looking at implementing d3 charts.  https://observablehq.com/@d3/grouped-bar-chart  So, rather than having matplotlib as a requirement and generating images, we can just do a query and pass json to the template of charts to generate.

janpascal commented on denyhosts_server/database.py:
```diff
@@ -145,6 +146,28 @@ def _evolve_database_v8(txn, dbtype):
+    elif dbtype == "MySQLdb":
+        autoincrement = "AUTO_INCREMENT"
+
+    if dbtype == "MySQLdb":
```
Please also create the table for sqlite databases

janpascal commented on denyhosts_server/database.py:
```diff
@@ -145,6 +146,28 @@ def _evolve_database_v8(txn, dbtype):
+        autoincrement = "AUTO_INCREMENT"
+
+    if dbtype == "MySQLdb":
+        txn.execute("""CREATE TABLE ClientVersions (
```
Please keep the table name convention (client_version)

janpascal commented on denyhosts_server/models.py:
```diff
@@ -18,20 +18,34 @@
 from twistar.dbobject import DBObject
 
+
+class ClientVersion(DBObject):
```
client_version

janpascal commented:
Hi Jose, thanks for your work. I think you forgot to include the changes to peering.py in the pull request, please add those, and also the unit tests.

josev814 committed:
resolving sqlite table creation

resolving table name convention matching

josev814 committed:
updating to query client_version table in it's own method

adding ClientVersion test

josev814 committed:
updating peering to include send_client_version update method

updating views to get the latest information from the db and passing that
     information to the peering servers

josev814 commented:
I made the suggested updates, and added in the missing peering code and unit tests.

The one issue I had is I wasn't able to get the unittests to run on my local environment.  I keep getting a six.moves module doesn't exist even though six is installed.  Trying to update the module seems to indicate that it's already up to date.

It would be nice if we could get the unittests moved over to travisci.
